### DISCUSSION TITLE
tokenizer_exceptions problem with Persian

### DISCUSSION THREAD

**Discussion by mosynaq** [explosion/spaCy#1772] [Language Support]
I am trying to train Persian to spaCy. One of the problems is here: in `tokenizer_exceptions.py`, spaCy expects concatenation of two orths, form the word itself like `do + n't = don't`, but for Persian, this expectation is not valid for some cases.
For example, the verb "بر نخواهد گشت" ( = s/he will not return), is made up of "بر" + "نـ" + "خواهد گشت".
("نـ" negates a Persian verb. Most of the times the negation thing comes in the beginning, but in some, cases like this one, it comes in between. )
As you can see you cannot simply concatenate orthes to form the full form.

Should spaCy expectation be changed? Or should I do something?

## My Environment

* **spaCy version:** 2.0.5
* **Platform:** Linux-4.11.0-14-generic-x86_64-with-LinuxMint-18.2-sonya
* **Models:** en, fa
* **Python version:** 3.5.2

**Comment by honnibal** [comment]
It's true that we don't really have a good solution to this for Persian, Arabic and other similar languages. I'm still not positive what the best strategy should be.
